File: Coding Sample/MS.NET/Fundamentals/Reflection/interval.cs
partial class Interval
{
	private int min;
	private int sec;

	partial void OnInit(); //partial method - maybe defined by other parts of this class

	public Interval(int m, int s)
	{
		min = m + s / 60;
		sec = s % 60;
		OnInit(); //will be removed if other part of this class does not define OnInit
	}

	public int Minutes
	{
		get {return min;}
	}

	public int Seconds
	{
		get {return sec;}
	}

	public int Time
	{
		get {return 60 * min + sec;}
	}

	public override string ToString()
	{
		if(sec < 10)
			return min + ":0" + sec;
		return min + ":" + sec;
	}

	public override int GetHashCode()
	{
		return 1000 * (60 * min + sec);
	}

	public override bool Equals(object other)
	{
		if(other is Interval)
		{
			Interval that = (Interval)other;
			return (this.min == that.min) && (this.sec == that.sec);			
		}

		return false;
	}
}

File: Coding Sample/Java/Fundamentals/Reflection/Interval.java
class Interval{

	private int minute;
	private int second;

	public Interval(int m, int s){
		minute = m + s / 60;
		second = s % 60;
	}

	public int minutes(){
		return minute;
	}

	public int seconds(){
		return second;
	}

	public int time(){
		return 60 * minute + second;
	}

	public Interval add(Interval other){
		return new Interval(minute + other.minute, second + other.second);
	}

	public String toString(){
		if(second < 10)
			return minute + ":0" + second;
		return minute + ":" + second;
	}

	public int hashCode(){
		return 1000 * time();
	}

	public boolean equals(Object other){
		if(other instanceof Interval){
			Interval that = (Interval)other;
			return (this.minute == that.minute) && (this.second == that.second);
		}
		return false;
	}
}

